libs/platform/src/lib/form/form-group/services/form-field-layout.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core';
+import { Observable, ReplaySubject } from 'rxjs';
+import { ColumnLayout } from '@fundamental-ngx/platform/shared';
+import { distinctUntilChanged, startWith } from 'rxjs/operators';
+import { normalizeColumnLayout } from '../helpers';
+
+type Layouts = {
+    labelColumnLayout: ColumnLayout;
+    fieldColumnLayout: ColumnLayout;
+    gapColumnLayout: ColumnLayout;
+};
+
+@Injectable()
+export class FormFieldLayoutService {
+    readonly needsInlineHelpPlace: Observable<boolean>;
+    private _needsInlineHelpPlace$ = new ReplaySubject<boolean>(1);
+    private _elementsMap: Map<Record<string, any>, boolean> = new Map();
+    private _needsInlineHelpPlace = false;
+
+    constructor() {
+        this.needsInlineHelpPlace = this._needsInlineHelpPlace$.pipe(startWith(false), distinctUntilChanged());
+    }
+
+    setNeedsInlineHelp(target: Record<string, any>, isNeeded: boolean): void {
+        this._elementsMap.set(target, isNeeded);
+        this._updateCombinedValue();
+    }
+
+    getFixedLayouts(layouts: Layouts): Layouts {
+        const { fieldColumnLayout, labelColumnLayout, gapColumnLayout } = {
+            fieldColumnLayout: normalizeColumnLayout({ ...layouts.fieldColumnLayout }),
+            labelColumnLayout: normalizeColumnLayout({ ...layouts.labelColumnLayout }),
+            gapColumnLayout: normalizeColumnLayout({ ...layouts.gapColumnLayout })
+        };
+        if (this._needsInlineHelpPlace) {
+            Object.keys(gapColumnLayout).forEach((sizeName) => {
+                if (sizeName !== 'S') {
+                    const shouldFitInOneLine =
+                        labelColumnLayout[sizeName] + fieldColumnLayout[sizeName] + gapColumnLayout[sizeName] <= 12;
```

Comment:
"magic" numbers. Maybe create constants for it or some configuration object that can also be injected here